base/env.jl

```diff
@@ -7,8 +7,16 @@ if Sys.iswindows()
     _hasenv(s::Vector{UInt16}) = _getenvlen(s) != 0 || Libc.GetLastError() != ERROR_ENVVAR_NOT_FOUND
     _hasenv(s::AbstractString) = _hasenv(cwstring(s))
 
+    const env_dict = IdDict{String, Vector{Cwchar_t}}()
+    const env_lock = ReentrantLock()
+
     function access_env(onError::Function, str::AbstractString)
-        var = cwstring(str)
+        var = get(env_dict, str, nothing)
+        if isnothing(var)
+            var = @lock env_lock begin
+                env_dict[str] = cwstring(str)
```

Comment:
Would it be useful to add a comment here explaining why we do this on Windows?

Alternatively, would it be sufficient to just add a comment linking the relevant GitHub issues and/or Discourse threads?